boards/native/board_init.c

```diff
@@ -20,7 +20,57 @@
 
 #ifdef MODULE_MTD
 #include "mtd_native.h"
+
+mtd_native_dev_t mtd0_dev = {
+    .dev = {
+        .driver = &native_flash_driver,
+        .sector_count = MTD_SECTOR_NUM,
+        .pages_per_sector = MTD_SECTOR_SIZE / MTD_PAGE_SIZE,
+        .page_size = MTD_PAGE_SIZE,
+    },
+    .fname = MTD_NATIVE_FILENAME,
+};
+
+mtd_dev_t *mtd0 = (mtd_dev_t *)&mtd0_dev;
+#endif
+
+#ifdef MODULE_VFS
+#include "vfs.h"
+
+/*
+ * On `native` we define auto-mounts for every file system.
+ *
+ * A 'real' board would typically always use the same file system to avoid
+ * data loss when re-formatting, but since `native` is for testing only we
+ * provide all file system definitions here.
+ */
+
+/* littlefs support */
+#if defined(MODULE_LITTLEFS)
```

Comment:
What happens if on native more than one file system is included? Such applications would, I presume, conflict right now (we don't have locks on MTD, so probably all but one mount would fail, but if autoformatting is on ... that might be fun).

Maybe it'd make sense to have native MTD devices for each file system (plus one that's not mounted), just to keep native board versatile.

Reply:
I would leave that to someone who actually wants to have multiple (auto-mounted) file systems on `native` :wink: 

One would have to edit the board definition to support multiple MTD devices (or partitions with `mtd_mapper`), so then the mount points could be edited too.